tests/core/OperationProcessor.spec.ts

```diff
@@ -693,10 +694,19 @@ describe('OperationProcessor', async () => {
           operationIndex: 2
         };
 
-        spyOn(DocumentComposer, 'applyUpdateOperation').and.throwError('Expected test error');
+        // Intentionally modifying the document before failing update patches to test original document is not modified.
+        spyOn(DocumentComposer, 'applyPatches').and.callFake((document, _patch) => {
+          document.publicKeys = [];
+          throw new Error('any error');
+        });
 
+        // The expected outcome of patch application failure (but all integrity checks including schema checks are passed )is:
+        // 1. Operation is considered successful.
+        // 2. Update commitment is updated/incremented.
+        // 3. DID document state remains the unchanged (same as prior to the patches being applied).
         const newDidState = await operationProcessor.apply(anchoredUpdateOperationModel, didState);
-        expect(newDidState!.lastOperationTransactionNumber).toEqual(1);
+        expect(newDidState!.lastOperationTransactionNumber).toEqual(2);
+        expect(newDidState!.nextUpdateCommitmentHash).toEqual(nextUpdateCommitment);
         expect(newDidState!.document).toBeDefined();
```

Comment:
We should probably expect it to be the same as a deep copy of the original document in didState.